modules/programs/hexchat.nix

```diff
@@ -0,0 +1,256 @@
+{ config, pkgs, lib, ... }:
+with builtins;
+with lib;
+let cfg = config.programs.hexchat;
+in {
+  meta.maintainers = [ maintainers.superherointj ];
+
+  options.programs.hexchat = {
+    channels = with lib.types;
+      let
+        modChannelOption = submodule {
+          options = {
+            options = let
+              channelOptions = submodule {
+                options = {
+                  autoconnect = mkOption {
+                    type = nullOr bool;
+                    description = "Autoconnect to network";
+                    default = false;
+                  };
+                  connect_to_selected_server_only = mkOption {
+                    type = nullOr bool;
+                    description = "Connect to selected server only";
+                    default = true;
+                  };
+                  bypass_proxy = mkOption {
+                    type = nullOr bool;
+                    description = "Bypass proxy";
+                    default = true;
+                  };
+                  use_ssl_for_all_servers = mkOption {
+                    type = nullOr bool;
+                    description = "Use SSL for all servers";
+                    default = false;
+                  };
+                  accept_invalid_ssl_certificates = mkOption {
+                    type = nullOr bool;
+                    description = "Accept invalid SSL certificates";
+                    default = false;
+                  };
+                  use_global_user_information = mkOption {
+                    type = nullOr bool;
+                    description = "Use global user information";
+                    default = false;
+                  };
+                };
+              };
+            in mkOption {
+              type = nullOr (channelOptions);
+              default = null;
+              description = "";
+              example = "";
+            };
+            nickname = mkOption {
+              type = str;
+              description = "Primary nickname";
+              defaultText = "nickname1";
+            };
+            nickname2 = mkOption {
+              type = str;
+              description = "Secondary nickname";
+              defaultText = "nickname2";
+            };
```

Comment:
I think there is also `nickname3` and `nickname4` (those are optional) :thinking: ?

Reply:
nickname3 is only available in global user information, and already can be configured in `conf` field as "irc_nick3". 
Channels don't have nickname3 in HexChat GUI interface. So it is not possible to track this field to it`s HexChat config value.
Nickname4 is unavailable globally and per channel in HexChat GUI.